### ISSUE TITLE
Failed on Nvidia GT 1030

### ISSUE THREAD

**Issue by jacen05** [Marvo2011/NVIDIA-vBIOS-VFIO-Patcher#3]
Hello,

Similar to #2, I got the following error when trying the patcher on the ROM available at https://www.techpowerup.com/vgabios/192799/Gigabyte.GT1030.2048.170417.rom :
```
# python3 nvidia_vbios_vfio_patcher.py -i Gigabyte.GT1030.2048.170417.rom -o Gigabyte.GT1030.2048.170417-patched.rom 
Opening the ROM file...
Scanning for ROM offsets...
ROM footer for RTX 2060 found!
Offsets found!
Running sanity checks...
Traceback (most recent call last):
  File "/root/NVIDIA-vBIOS-VFIO-Patcher/nvidia_vbios_vfio_patcher.py", line 213, in <module>
    main()
  File "/root/NVIDIA-vBIOS-VFIO-Patcher/nvidia_vbios_vfio_patcher.py", line 191, in main
    rom.run_sanity_tests(args.ignore_sanity_check)
  File "/root/NVIDIA-vBIOS-VFIO-Patcher/nvidia_vbios_vfio_patcher.py", line 105, in run_sanity_tests
    raise CheckException(
__main__.CheckException: Expected only one 'NPDS' marker between header and footer, found 3
```

I have no knowledge about card bios and patching stuff however I can do some testing if it can help to debug the issue.
Obviously, for Proxmox GPU passthrough, I tried first without specifying the ROM, and with specifying the standard (unpatched) ROM file. No success in both cases (error 43 as usual).

**Comment by jacen05** [comment]
Update: as the card was detected as RTX 2060 instead of GT 1030, I commented out the regex line for RTX 2060 [here](https://github.com/Marvo2011/NVIDIA-vBIOS-VFIO-Patcher/blob/master/nvidia_vbios_vfio_patcher.py#L71), and then it was detected as GTX 10XX. However, when using the patched ROM, Windows does not start (it starts correctly if I set the path to the unpatched ROM).

I also tried using the --disable-footer-strip option, but it produces the same erroneous behavior.

What can I try next?

**Comment by jacen05** [comment]
Update 2: I got this working by extracting the GPU BIOS with nvflash. It seems that BIOS for GT 1030 from [TechPowerUp](https://www.techpowerup.com/vgabios/192799/Gigabyte.GT1030.2048.170417.rom) is incorrect. However as noted in my previous comment I had to comment out [the regex for RTX 2060](https://github.com/Marvo2011/NVIDIA-vBIOS-VFIO-Patcher/blob/master/nvidia_vbios_vfio_patcher.py#L71) as the GT 1030 was detected as RTX 2060 by the script.

Now I have a working passthrough solution, thanks!
